### PR TITLE
[IMG-114] Align remote docker docs with new tagging system

### PR DESCRIPTION
**Pull request by JalexChen**
# Description
Align and update docs according to the tags specified in the new remote docker image policy

# Reasons
https://circleci.atlassian.net/jira/software/c/projects/IMG/boards/657?selectedIssue=IMG-114

# Content Checklist
Please follow our style when contributing to CircleCI docs. Our style guide is here: [https://circleci.com/docs/style/style-guide-overview](https://circleci.com/docs/style/style-guide-overview).

Please take a moment to check through the following items when submitting your PR (this is just a guide so will not be relevant for all PRs) 😸:

- [x] Break up walls of text by adding paragraph breaks.
- [x] Consider if the content could benefit from more structure, such as lists or tables, to make it easier to consume.
- [x] Keep the title between 20 and 70 characters.
- [x] Consider whether the content would benefit from more subsections (h2-h6 headings) to make it easier to consume.
- [x] Check all headings h1-h6 are in sentence case (only first letter is capitalized).
- [N/A] Is there a "Next steps" section at the end of the page giving the reader a clear path to what to read next?
- [x] Include relevant backlinks to other CircleCI docs/pages.

### CODE REVIEW

**Comment by JalexChen** [comment]
@sebastian-lerner can you confirm if we intend to support `docker24` as a tag for customers to use?

**Comment by JalexChen** [comment]
i don't think it was clear

In the plan: Moved from `current`, `edge` and `previous` tags `docker24` tag.

Should it be: Moved from XXX _to_ `docker24` tag?

**Comment by sebastian-lerner** [comment]
@brentmmarks what should the plan say?

**Comment by JalexChen** [comment]
i think what i mentioned should be what the plan says. it's clear and limits customer options for default to `current` and `version`= blank. `docker24` becomes an option once a new major docker version is released, which would be consistent with the rest of the policy

**Comment by brentmmarks** [comment]
We should remove that alias if no customers are using it if this is plan - which I am on-board with. Plan currently states what you mentioned Jeff

File: jekyll/_cci2/remote-docker-images-support-policy.adoc [hunk 1]
```diff
@@ -35,19 +35,19 @@ We support various tags for the remote Docker environment to allow you to choose
 
 For the latest major version of Docker:
```
Comment (brentmmarks) [inline jekyll/_cci2/remote-docker-images-support-policy.adoc:36]: We should confirm with support updated policy before touching this multiple times

**Comment by brentmmarks** [comment]
Please rebase (once https://github.com/circleci/circleci-docs/pull/8439 is merged) with changes sebastian made to show what docker version is on default

File: jekyll/_cci2/remote-docker-images-support-policy.adoc [hunk 2]
```diff
@@ -35,22 +35,18 @@ We support various tags for the remote Docker environment to allow you to choose
...
```
Comment (brentmmarks) [inline jekyll/_cci2/remote-docker-images-support-policy.adoc:36]: Overview needs to be fixed to match new policy with some minor changes:

`This document outlines the [CircleCI remote Docker image](https://circleci.com/docs/building-docker-images/) release, update, and deprecation policy. This policy applies to all CircleCI remote Docker images built for the remote Docker feature (setup_remote_docker).`

**Review by brentmmarks — APPROVED** [review]

**Review by felicianotech — APPROVED** [review]

**Pull request merged**